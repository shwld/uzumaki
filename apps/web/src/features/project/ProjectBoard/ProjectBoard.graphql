fragment ProjectBoard_Story on Story {
  id
  kind
  title
  state
  position
  priority
  points
  isDeleted
  isUnEstimated
  isCompleted
  completedAt
  projectId
}

fragment ProjectBoard_Project on Project {
  id
  boardConfig {
    id
    startOn
    startIterationWeekNumber
    iterationLength
  }
  boardStatus {
    id
    velocity
  }
}

query ProjectBoard_Status($projectId: ID!) {
  viewer {
    id
    project(id: $projectId) {
      ...ProjectBoard_Project
    }
  }
}
query ProjectBoard_Stories(
  $projectId: ID!
  $position: ProjectStoriesSearchPosition
  $cursor: String
) {
  viewer {
    id
    project(id: $projectId) {
      id
      stories(position: $position, first: 50, after: $cursor) {
        edges {
          node {
            ...ProjectBoard_Story
          }
          cursor
        }
        pageInfo {
          hasNextPage
          endCursor
        }
      }
    }
  }
}

mutation ProjectBoard_MoveStories($input: MoveStoriesInput!) {
  moveStories(input: $input) {
    ... on MoveStoriesSuccessResult {
      result {
        id
        position
        priority
      }
    }
    ... on InvalidArgumentsResult {
      issues {
        field
        message
      }
    }
  }
}

subscription ProjectBoard_Subsc($projectId: ID!) {
  subscribeStoryUpdate(projectId: $projectId) {
    id
  }
}
