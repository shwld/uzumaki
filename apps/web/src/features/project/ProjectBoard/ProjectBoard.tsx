import { HStack } from '@chakra-ui/react';
import { FC, useMemo, useState } from 'react';
import { useMovableStoryList } from './hooks/useMovableStoryList';
import {
  ProjectBoard_ProjectFragment,
  ProjectBoard_StoryFragment,
  useProjectBoard_StatusQuery,
  useProjectBoard_SubscSubscription,
} from './ProjectBoard.generated';
import { DoneBoard } from './boards/DoneBoard';
import { DragDropContext } from 'react-beautiful-dnd';
import { CurrentBoard } from './boards/CurrentBoard';
import { BacklogBoard } from './boards/BacklogBoard';
import { IceboxBoard } from './boards/IceboxBoard';
import { useExtendedProjectBoardQuery } from './hooks/useExtendedProjectBoardQuery';
import dayjs from 'dayjs';
import { InspectVelocityButton } from './components/InspectVelocityButton';

export const ProjectBoard: FC<{
  projectId: string;
}> = ({ projectId }) => {
  const [{ fetching, error, data }] = useProjectBoard_StatusQuery({
    variables: { projectId },
  });
  const [res] = useProjectBoard_SubscSubscription(
    {
      variables: {
        projectId,
      },
    },
    (messages: any, message) => {
      console.log({ messages, message });
      return [message];
    }
  );
  // console.log(res.data);
  const project = data?.viewer?.project;

  if (fetching) return <></>;
  if (error) return <></>;
  if (project == null) return <></>;

  return <StoriesContainer project={project} />;
};

/**
 * PRIVATE
 */
const StoriesContainer: FC<{
  project: ProjectBoard_ProjectFragment;
}> = ({ project }) => {
  const result = useExtendedProjectBoardQuery(project.id);

  if (result.fetching) return <></>;
  if (result.error) return <></>;

  return <ProjectStoryBoards project={project} stories={result.stories} />;
};

const ProjectStoryBoards: FC<{
  project: ProjectBoard_ProjectFragment;
  stories: ProjectBoard_StoryFragment[];
}> = ({ project, stories }) => {
  const projectStartDate = useMemo(
    () => dayjs().day(project.boardConfig.startIterationWeekNumber).toDate(),
    [project]
  );
  const [currentVelocity, setCurrentVelocity] = useState(
    project.boardStatus.velocity
  );
  const {
    currentStories,
    backlogStories,
    iceboxStories,
    doneStories,
    handleDragEnd,
  } = useMovableStoryList(project.id, projectStartDate, stories);
  return (
    <HStack align="stretch" h="calc(100vh - 5rem)">
      <DragDropContext onDragEnd={handleDragEnd}>
        <DoneBoard
          stories={doneStories.done}
          iterationLength={project.boardConfig.iterationLength}
          currentIterationStartDate={projectStartDate}
        />
        <CurrentBoard
          projectId={project.id}
          currentVelocity={currentVelocity}
          iterationLengthInWeek={project.boardConfig.iterationLength}
          iterationStartDate={projectStartDate}
          stories={currentStories}
          doneStories={doneStories.current}
          header={
            <InspectVelocityButton
              persistedVelocity={project.boardStatus.velocity}
              onChangeVelocity={setCurrentVelocity}
            />
          }
        />
        <BacklogBoard
          projectId={project.id}
          currentVelocity={currentVelocity}
          iterationLengthInWeek={project.boardConfig.iterationLength}
          currentIterationStartDate={projectStartDate}
          stories={backlogStories}
        />
        <IceboxBoard projectId={project.id} stories={iceboxStories} />
      </DragDropContext>
    </HStack>
  );
};
