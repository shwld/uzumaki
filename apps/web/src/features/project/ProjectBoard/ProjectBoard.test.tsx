import { ProjectBoard } from './ProjectBoard';
import { render } from '@testing-library/react';
import { MockedUrqlProvider } from '~/test/MockedUrqlProvider';

describe('ProjectBoard', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedUrqlProvider>
        <ProjectBoard projectId="test" />
      </MockedUrqlProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('Current')).toBeTruthy();
  });
});
