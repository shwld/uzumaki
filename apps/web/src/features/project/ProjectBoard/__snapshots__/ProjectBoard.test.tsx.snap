// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectBoard Snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-op5fxw"
  >
    <div
      class="css-14fuj5p"
    >
      <ul
        class="css-1v6xhwy"
        role="list"
      >
        <li
          class="css-wzbbt3"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <button
                aria-label=""
                class="chakra-button css-198bfjk"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-zmpx8u"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
            <p
              class="chakra-text css-1dfa66e"
            >
              Done
            </p>
          </div>
        </li>
        <li
          class="css-lybpxv"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <svg
                class="chakra-icon css-17hpw6g"
                focusable="false"
                viewBox="0 0 14 14"
              >
                <path
                  d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
                  fill="currentColor"
                />
              </svg>
              <p
                class="chakra-text css-1aq3av"
              >
                 points
              </p>
            </div>
            <div
              class="chakra-stack css-84zodg"
            >
              <p
                class="chakra-text css-1aq3av"
              >
                20 Nov - 10 Jul
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="css-14fuj5p"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="CURRENT"
    >
      <ul
        class="css-155za0w"
        role="list"
      >
        <li
          class="css-wzbbt3"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <button
                aria-label=""
                class="chakra-button css-198bfjk"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-zmpx8u"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <button
                class="chakra-button css-c3joui"
                type="button"
              >
                <span
                  class="chakra-button__icon css-1wh2kri"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-icon css-uqqaxt"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 2a.5.5 0 0 1 .5.5V4a.5.5 0 0 1-1 0V2.5A.5.5 0 0 1 8 2zM3.732 3.732a.5.5 0 0 1 .707 0l.915.914a.5.5 0 1 1-.708.708l-.914-.915a.5.5 0 0 1 0-.707zM2 8a.5.5 0 0 1 .5-.5h1.586a.5.5 0 0 1 0 1H2.5A.5.5 0 0 1 2 8zm9.5 0a.5.5 0 0 1 .5-.5h1.5a.5.5 0 0 1 0 1H12a.5.5 0 0 1-.5-.5zm.754-4.246a.389.389 0 0 0-.527-.02L7.547 7.31A.91.91 0 1 0 8.85 8.569l3.434-4.297a.389.389 0 0 0-.029-.518z"
                    />
                    <path
                      d="M6.664 15.889A8 8 0 1 1 9.336.11a8 8 0 0 1-2.672 15.78zm-4.665-4.283A11.945 11.945 0 0 1 8 10c2.186 0 4.236.585 6.001 1.606a7 7 0 1 0-12.002 0z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                9707
              </button>
              <button
                class="chakra-button css-c3joui"
                type="button"
              >
                <svg
                  class="chakra-icon chakra-icon css-ikjmp3"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                    fill="currentColor"
                  />
                </svg>
                Add story
              </button>
            </div>
            <p
              class="chakra-text css-1dfa66e"
            >
              Current Iteration
            </p>
          </div>
        </li>
        <li
          class="css-lybpxv"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <svg
                class="chakra-icon css-17hpw6g"
                focusable="false"
                viewBox="0 0 14 14"
              >
                <path
                  d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
                  fill="currentColor"
                />
              </svg>
              <p
                class="chakra-text css-1aq3av"
              >
                0 points
              </p>
            </div>
            <div
              class="chakra-stack css-84zodg"
            >
              <p
                class="chakra-text css-1aq3av"
              >
                18 Mar - 05 Nov
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="css-14fuj5p"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="BACKLOG"
    >
      <ul
        class="css-155za0w"
        role="list"
      >
        <li
          class="css-wzbbt3"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <button
                aria-label=""
                class="chakra-button css-198bfjk"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-zmpx8u"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <button
                class="chakra-button css-c3joui"
                type="button"
              >
                <svg
                  class="chakra-icon chakra-icon css-ikjmp3"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                    fill="currentColor"
                  />
                </svg>
                Add story
              </button>
            </div>
            <p
              class="chakra-text css-1dfa66e"
            >
              Backlog
            </p>
          </div>
        </li>
        <li
          class="css-lybpxv"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <svg
                class="chakra-icon css-17hpw6g"
                focusable="false"
                viewBox="0 0 14 14"
              >
                <path
                  d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
                  fill="currentColor"
                />
              </svg>
              <p
                class="chakra-text css-1aq3av"
              >
                12000 points
              </p>
            </div>
            <div
              class="chakra-stack css-84zodg"
            >
              <p
                class="chakra-text css-1aq3av"
              >
                05 Nov - 27 Jun
              </p>
            </div>
          </div>
        </li>
        <div
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <li
            class="css-zazzmu"
          >
            <div
              class="chakra-stack css-1oeb4ru"
            >
              <div
                class="chakra-stack css-yve0xp"
              >
                <svg
                  class="chakra-icon css-1c0gt8a"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M304 280h416c4.4 0 8-3.6 8-8 0-40-8.8-76.7-25.9-108.1-17.2-31.5-42.5-56.8-74-74C596.7 72.8 560 64 520 64h-16c-40 0-76.7 8.8-108.1 25.9-31.5 17.2-56.8 42.5-74 74C304.8 195.3 296 232 296 272c0 4.4 3.6 8 8 8z"
                  />
                  <path
                    d="M940 512H792V412c76.8 0 139-62.2 139-139 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 34.8-28.2 63-63 63H232c-34.8 0-63-28.2-63-63 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 76.8 62.2 139 139 139v100H84c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h148v96c0 6.5.2 13 .7 19.3C164.1 728.6 116 796.7 116 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-44.2 23.9-82.9 59.6-103.7 6 17.2 13.6 33.6 22.7 49 24.3 41.5 59 76.2 100.5 100.5 28.9 16.9 61 28.8 95.3 34.5 4.4 0 8-3.6 8-8V484c0-4.4 3.6-8 8-8h60c4.4 0 8 3.6 8 8v464.2c0 4.4 3.6 8 8 8 34.3-5.7 66.4-17.6 95.3-34.5 41.5-24.3 76.2-59 100.5-100.5 9.1-15.5 16.7-31.9 22.7-49C812.1 793.1 836 831.8 836 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-79.3-48.1-147.4-116.7-176.7.4-6.4.7-12.8.7-19.3v-96h148c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
                  />
                </svg>
                <p
                  class="chakra-text css-1bvg7gj"
                />
                <p
                  class="chakra-text css-504jus"
                >
                  deleniti
                </p>
              </div>
              <div
                class="chakra-stack css-2nt6bn"
              />
            </div>
          </li>
        </div>
        <div
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <li
            class="css-zazzmu"
          >
            <div
              class="chakra-stack css-1oeb4ru"
            >
              <div
                class="chakra-stack css-yve0xp"
              >
                <svg
                  class="chakra-icon css-1c0gt8a"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M304 280h416c4.4 0 8-3.6 8-8 0-40-8.8-76.7-25.9-108.1-17.2-31.5-42.5-56.8-74-74C596.7 72.8 560 64 520 64h-16c-40 0-76.7 8.8-108.1 25.9-31.5 17.2-56.8 42.5-74 74C304.8 195.3 296 232 296 272c0 4.4 3.6 8 8 8z"
                  />
                  <path
                    d="M940 512H792V412c76.8 0 139-62.2 139-139 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 34.8-28.2 63-63 63H232c-34.8 0-63-28.2-63-63 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 76.8 62.2 139 139 139v100H84c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h148v96c0 6.5.2 13 .7 19.3C164.1 728.6 116 796.7 116 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-44.2 23.9-82.9 59.6-103.7 6 17.2 13.6 33.6 22.7 49 24.3 41.5 59 76.2 100.5 100.5 28.9 16.9 61 28.8 95.3 34.5 4.4 0 8-3.6 8-8V484c0-4.4 3.6-8 8-8h60c4.4 0 8 3.6 8 8v464.2c0 4.4 3.6 8 8 8 34.3-5.7 66.4-17.6 95.3-34.5 41.5-24.3 76.2-59 100.5-100.5 9.1-15.5 16.7-31.9 22.7-49C812.1 793.1 836 831.8 836 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-79.3-48.1-147.4-116.7-176.7.4-6.4.7-12.8.7-19.3v-96h148c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
                  />
                </svg>
                <p
                  class="chakra-text css-1bvg7gj"
                />
                <p
                  class="chakra-text css-504jus"
                >
                  deleniti
                </p>
              </div>
              <div
                class="chakra-stack css-2nt6bn"
              />
            </div>
          </li>
        </div>
        <div
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <li
            class="css-zazzmu"
          >
            <div
              class="chakra-stack css-1oeb4ru"
            >
              <div
                class="chakra-stack css-yve0xp"
              >
                <svg
                  class="chakra-icon css-1c0gt8a"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M304 280h416c4.4 0 8-3.6 8-8 0-40-8.8-76.7-25.9-108.1-17.2-31.5-42.5-56.8-74-74C596.7 72.8 560 64 520 64h-16c-40 0-76.7 8.8-108.1 25.9-31.5 17.2-56.8 42.5-74 74C304.8 195.3 296 232 296 272c0 4.4 3.6 8 8 8z"
                  />
                  <path
                    d="M940 512H792V412c76.8 0 139-62.2 139-139 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 34.8-28.2 63-63 63H232c-34.8 0-63-28.2-63-63 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 76.8 62.2 139 139 139v100H84c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h148v96c0 6.5.2 13 .7 19.3C164.1 728.6 116 796.7 116 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-44.2 23.9-82.9 59.6-103.7 6 17.2 13.6 33.6 22.7 49 24.3 41.5 59 76.2 100.5 100.5 28.9 16.9 61 28.8 95.3 34.5 4.4 0 8-3.6 8-8V484c0-4.4 3.6-8 8-8h60c4.4 0 8 3.6 8 8v464.2c0 4.4 3.6 8 8 8 34.3-5.7 66.4-17.6 95.3-34.5 41.5-24.3 76.2-59 100.5-100.5 9.1-15.5 16.7-31.9 22.7-49C812.1 793.1 836 831.8 836 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-79.3-48.1-147.4-116.7-176.7.4-6.4.7-12.8.7-19.3v-96h148c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
                  />
                </svg>
                <p
                  class="chakra-text css-1bvg7gj"
                />
                <p
                  class="chakra-text css-504jus"
                >
                  deleniti
                </p>
              </div>
              <div
                class="chakra-stack css-2nt6bn"
              />
            </div>
          </li>
        </div>
        <li
          class="css-lybpxv"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <svg
                class="chakra-icon css-17hpw6g"
                focusable="false"
                viewBox="0 0 14 14"
              >
                <path
                  d="M10.8889,5.5 L3.11111,5.5 L3.11111,7.05556 L10.8889,7.05556 L10.8889,5.5 Z M12.4444,1.05556 L11.6667,1.05556 L11.6667,0 L10.1111,0 L10.1111,1.05556 L3.88889,1.05556 L3.88889,0 L2.33333,0 L2.33333,1.05556 L1.55556,1.05556 C0.692222,1.05556 0.00777777,1.75556 0.00777777,2.61111 L0,12.5 C0,13.3556 0.692222,14 1.55556,14 L12.4444,14 C13.3,14 14,13.3556 14,12.5 L14,2.61111 C14,1.75556 13.3,1.05556 12.4444,1.05556 Z M12.4444,12.5 L1.55556,12.5 L1.55556,3.94444 L12.4444,3.94444 L12.4444,12.5 Z M8.55556,8.61111 L3.11111,8.61111 L3.11111,10.1667 L8.55556,10.1667 L8.55556,8.61111 Z"
                  fill="currentColor"
                />
              </svg>
              <p
                class="chakra-text css-1aq3av"
              >
                6293 points
              </p>
            </div>
            <div
              class="chakra-stack css-84zodg"
            >
              <p
                class="chakra-text css-1aq3av"
              >
                27 Jun - 15 Feb
              </p>
            </div>
          </div>
        </li>
        <div
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="43319be6-5d3b-4a00-8485-e4f526c6b331"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <li
            class="css-zazzmu"
          >
            <div
              class="chakra-stack css-1oeb4ru"
            >
              <div
                class="chakra-stack css-yve0xp"
              >
                <svg
                  class="chakra-icon css-1c0gt8a"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M304 280h416c4.4 0 8-3.6 8-8 0-40-8.8-76.7-25.9-108.1-17.2-31.5-42.5-56.8-74-74C596.7 72.8 560 64 520 64h-16c-40 0-76.7 8.8-108.1 25.9-31.5 17.2-56.8 42.5-74 74C304.8 195.3 296 232 296 272c0 4.4 3.6 8 8 8z"
                  />
                  <path
                    d="M940 512H792V412c76.8 0 139-62.2 139-139 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 34.8-28.2 63-63 63H232c-34.8 0-63-28.2-63-63 0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 76.8 62.2 139 139 139v100H84c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h148v96c0 6.5.2 13 .7 19.3C164.1 728.6 116 796.7 116 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-44.2 23.9-82.9 59.6-103.7 6 17.2 13.6 33.6 22.7 49 24.3 41.5 59 76.2 100.5 100.5 28.9 16.9 61 28.8 95.3 34.5 4.4 0 8-3.6 8-8V484c0-4.4 3.6-8 8-8h60c4.4 0 8 3.6 8 8v464.2c0 4.4 3.6 8 8 8 34.3-5.7 66.4-17.6 95.3-34.5 41.5-24.3 76.2-59 100.5-100.5 9.1-15.5 16.7-31.9 22.7-49C812.1 793.1 836 831.8 836 876c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8 0-79.3-48.1-147.4-116.7-176.7.4-6.4.7-12.8.7-19.3v-96h148c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
                  />
                </svg>
                <p
                  class="chakra-text css-1bvg7gj"
                />
                <p
                  class="chakra-text css-504jus"
                >
                  deleniti
                </p>
              </div>
              <div
                class="chakra-stack css-2nt6bn"
              />
            </div>
          </li>
        </div>
      </ul>
    </div>
    <div
      class="css-14fuj5p"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="ICEBOX"
    >
      <ul
        class="css-155za0w"
        role="list"
      >
        <li
          class="css-wzbbt3"
        >
          <div
            class="chakra-stack css-1oeb4ru"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <button
                aria-label=""
                class="chakra-button css-198bfjk"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-zmpx8u"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <button
                class="chakra-button css-c3joui"
                type="button"
              >
                <svg
                  class="chakra-icon chakra-icon css-ikjmp3"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                    fill="currentColor"
                  />
                </svg>
                Add story
              </button>
            </div>
            <p
              class="chakra-text css-1dfa66e"
            >
              Icebox
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
